src/utils/MySQLDataHandler.py

```diff
@@ -43,6 +43,66 @@ def base64decode(str_val):
     return base64.b64decode(str_val.encode("utf-8")).decode("utf-8")
 
 
+def default_vc_entries(config):
+    if "worker_sku_cnt" not in config or "sku_mapping" not in config:
+        print("Warning: no default value would be added to VC table. Need to manually specify")
+        return ""
+
+    worker_sku_cnt, sku_mapping = config["worker_sku_cnt"], config["sku_mapping"]
+    quota_dict = {}
+    old_meta = {}
+    resource_quota = {"cpu":{}, "memory":{}, "gpu":{}, "gpu_memory":{}}
+    for sku, cnt in worker_sku_cnt.items():
+        gpu_type = sku_mapping.get(sku, {}).get("gpu-type", "None")
+        num_gpu_per_node = sku_mapping.get(sku, {}).get("gpu", 0)
+        quota_dict[gpu_type] = quota_dict.get(gpu_type, 0) + cnt * num_gpu_per_node
+        old_meta[gpu_type] = num_gpu_per_node
+        sku_name_in_map = sku if sku in sku_mapping else ""
+        meta_tmp = sku_mapping.get(sku_name_in_map, {})
+        for r_type in resource_quota.keys():
+            resource_quota[r_type][sku_name_in_map] = resource_quota[r_type].get(sku_name_in_map, 0) + meta_tmp.get(r_type, 0) * cnt
+
+    for r_type in ["cpu", "memory"]:
+        for sku, val in resource_quota[r_type].items():
+            resource_quota[r_type][sku] *= 0.9
+
+    # if a string would be formatted, then need to escape { by using {{
+    name_and_par = "AS SELECT \'{}\' AS vcName, NULL AS parent".format(config['defalt_virtual_cluster_name'])
+    quota = "'{" + ",".join(['"{}":{}'.format(ky, cnt) for ky, cnt in quota_dict.items()]) + "}' AS quota"
+    metadata = "'{" + ",".join(['"{}":{{"num_gpu_per_node":{}}}'.format(ky, cnt) for ky, cnt in old_meta.items()]) + "}' AS metadata"
+    res_quota = "'{"
+    res_quota_lst = []
+    for res, res_q in resource_quota.items():
+        quot_str = '"{}":{{'.format(res)
```

Comment:
Manual construction is hard to maintain. You can construct `resource_quota` dictionary and `json.dumps(resource_quota, separators=(",",":"))`.